torch/optim/adamax.py

```diff
@@ -336,19 +352,35 @@ def _multi_tensor_adamax(
             else:
                 grouped_grads = torch._foreach_add(grouped_grads, grouped_params, alpha=weight_decay)
 
+
         # Update biased first moment estimate.
         torch._foreach_lerp_(grouped_exp_avgs, grouped_grads, 1 - beta1)
 
         # Update the exponentially weighted infinity norm.
         torch._foreach_mul_(grouped_exp_infs, beta2)
 
-        for exp_inf, grad in zip(grouped_exp_infs, grouped_grads):
-            norm_buf = torch.cat(
-                [exp_inf.unsqueeze(0), grad.abs().add_(eps).unsqueeze_(0)], 0
-            )
-            torch.max(norm_buf, 0, keepdim=False, out=(exp_inf, exp_inf.new().long()))
+        # in this case, we need to introduce a copy of the grads
+        # since one has not been introduced previously
+        if not maximize and weight_decay == 0:
+            grouped_grads = torch._foreach_abs(grouped_grads)
+        else:
+            torch._foreach_abs_(grouped_grads)
 
-        bias_corrections = [1 - beta1 ** _get_value(step) for step in grouped_state_steps]
-        clr = _stack_if_compiling([-1 * (lr / bias_correction) for bias_correction in bias_corrections])
+        torch._foreach_add_(grouped_grads, eps)
+        torch._foreach_maximum_(grouped_exp_infs, grouped_grads)
 
-        torch._foreach_addcdiv_(grouped_params, grouped_exp_avgs, grouped_exp_infs, clr)
+        if capturable:
+            bias_corrections = torch._foreach_pow(beta1, grouped_state_steps)
+            # foreach_sub doesn't allow a scalar as the first arg
+            torch._foreach_sub_(bias_corrections, 1)
+
+            # foreach_div doesn't allow a scalar as the first arg
+            torch._foreach_div_(bias_corrections, lr)
+            torch._foreach_reciprocal_(bias_corrections)
+
+            numerator = torch._foreach_mul(grouped_exp_avgs, bias_corrections)
+            torch._foreach_addcdiv_(grouped_params, numerator, grouped_exp_infs)
+        else:
+            bias_corrections = [1 - beta1 ** _get_value(step) for step in grouped_state_steps]
+            step_size = _stack_if_compiling([(lr / bc) * -1 for bc in bias_corrections])
```

Comment:
Do we still need the stack if compiling here? this path shouldn't be run by compile right?